services/device-registry-base/src/main/java/org/eclipse/hono/service/management/device/TemplateHelper.java

```diff
@@ -0,0 +1,145 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.hono.service.management.device;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Pattern;
+
+import javax.naming.InvalidNameException;
+import javax.naming.ldap.LdapName;
+import javax.naming.ldap.Rdn;
+
+import org.eclipse.hono.util.RegistryManagementConstants;
+import org.eclipse.hono.util.Strings;
+
+/**
+ * A utility class for handling template used for generating device identifier during auto-provisioning.
+ */
+public final class TemplateHelper {
+
+    private static final String QUOTED_PLACEHOLDER_SUBJECT_DN = Pattern
+            .quote(RegistryManagementConstants.PLACEHOLDER_SUBJECT_DN);
+
+    private TemplateHelper() {
```

Comment:
wouldn't it make sense to turn this into an `IdentityTemplate` class and create instances of it using the template string in the constructor? The *fill* method would then only accept a DN as the argument, right? It might be worth considering to rename the *fill* method to something like *apply* as we apply the DN to the template, producing the identity.

Reply:
It makes sense. Since we name our classes normally using a noun, how about `TemplateIdentifer`?

Reply:
I have pushed a commit with respective changes. Would you mind taking a look again?